### PR TITLE
Consent Privacy Request Status when Identity Verification Required

### PR DESCRIPTION
**Pull request by pattisdr**
Closes https://github.com/ethyca/fides/issues/2735

### Code Changes

* [ ] Update the logic as to which initial status a Privacy Request is given.  If the request is already authenticated, even if subject_identity_verification_required=True, this should be a `PENDING` request, not "identity unverified".  This was putting the consent request in a non-executable state. It's identity was already verified, but the privacy request is sitting there in the wrong state and can't be manually approved.

### Steps to Confirm

* [x]  Set your `src/fides/data/test_env/fides.test_env.toml` file to have `subject_identity_verification_required = true`
* [x] Update one consent option in the test env config.json to be executable. Rebuild the privacy center
* [x]  Run `nox -s fides_env(dev)`
* [x] Setup mailgun locally
* [x] Go to the privacy center. Do a hard refresh
* [x] Click on Manage your consent and enter an email you control
* [x] Go to your email and retrieve the verification code
* [x] Return to the privacy center and enter the code into the `Enter verification code` modal and click submit
* [x] Update your consent preferences and click save
* [x] Login to the admin app.  You should see the associated privacy request with a status of "New" instead of "Identity Unverified"

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [ ] Update `CHANGELOG.md`

### Description Of Changes

 When updating consent preferences, and subject_identity_verification_required is True, Privacy Request should be put in a status of "pending" not "identity_unverified" as identity was verified before the consent preferences were saved.

Screenshot 2023-03-02 at 2 42 14 PM

### CODE REVIEW

File: src/fides/api/ops/api/v1/endpoints/drp_endpoints.py [hunk 1]
```diff
@@ -93,7 +93,10 @@ async def create_drp_privacy_request(
         )
 
     privacy_request_kwargs: Dict[str, Any] = build_required_privacy_request_kwargs(
-        None, policy.id, config_proxy.execution.subject_identity_verification_required
+        None,
+        policy.id,
+        config_proxy.execution.subject_identity_verification_required,
+        authenticated=False,
```
Comment (pattisdr) [inline src/fides/api/ops/api/v1/endpoints/drp_endpoints.py:99]: Giving the dsr privacy requests here a default authenticated value of False

File: src/fides/api/ops/api/v1/endpoints/privacy_request_endpoints.py [hunk 2]
```diff
@@ -1633,6 +1633,7 @@ def create_privacy_request_func(
             privacy_request_data.requested_at,
             policy.id,
             config_proxy.execution.subject_identity_verification_required,
+            authenticated,
```
Comment (pattisdr) [inline src/fides/api/ops/api/v1/endpoints/privacy_request_endpoints.py:1636]: When privacy requests are created from the workflow of updating consent preferences, they are automatically treated as authenticated as they would have been authenticated via the consent request workflow.  Passing in that authenticated value here lets us put the privacy request in the correct state.

File: src/fides/api/ops/service/privacy_request/request_service.py [hunk 3]
```diff
@@ -30,7 +31,7 @@ def build_required_privacy_request_kwargs(
     """
     status = (
         PrivacyRequestStatus.identity_unverified
-        if verification_required
+        if verification_required and not authenticated
```
Comment (pattisdr) [inline src/fides/api/ops/service/privacy_request/request_service.py:34]: Primary change of this PR

File: tests/fixtures/saas/vend_fixtures.py [hunk 4]
```diff
@@ -121,13 +119,15 @@ def vend_create_erasure_data(
...
     }
 
-    customers_response = requests.post(url=f"{base_url}/api/2.0/customers", headers=headers, json=body)
+    customers_response = requests.post(
+        url=f"{base_url}/api/2.0/customers", headers=headers, json=body
+    )
     customer = customers_response.json()
 
-    # there is no endpoint to create a sale so we cannot test it 
+    # there is no endpoint to create a sale so we cannot test it
```
Comment (pattisdr) [inline tests/fixtures/saas/vend_fixtures.py:130]: All black formatting

File: tests/ops/api/v1/endpoints/test_consent_request_endpoints.py [hunk 5]
```diff
@@ -808,6 +809,63 @@ def test_set_consent_consent_preferences(
...
+        response = api_client.patch(
+            f"{V1_URL_PREFIX}{CONSENT_REQUEST_PREFERENCES_WITH_ID.format(consent_request_id=consent_request.id)}",
+            json=data,
+        )
+
+        assert response.status_code == 200
+
+        mock_verify_identity.assert_called_with(verification_code)
+        db.refresh(consent_request)
+        assert consent_request.privacy_request.status == PrivacyRequestStatus.pending
```
Comment (pattisdr) [inline tests/ops/api/v1/endpoints/test_consent_request_endpoints.py:866]: This test would have had a status of "identity unverified" before.

File: tests/ops/service/privacy_request/test_request_service.py [hunk 6]
```diff
@@ -108,3 +111,40 @@ async def test_poll_server_for_completion_non_200(async_api_client, db, policy):
             client=async_api_client,
             poll_interval_seconds=1,
         )
+
+
+class TestBuildPrivacyRequestRequiredKwargs:
```
Comment (seanpreston) [inline tests/ops/service/privacy_request/test_request_service.py:116]: Thankyou for adding these extras! 🙏

**Review by seanpreston — APPROVED** [review]
Thankyou @pattisdr

**Pull request merged**